### ISSUE TITLE
Initial Status bar color will change when applyying appBarTheme in next the route and popping to initial page on iOS

### ISSUE THREAD

**Issue by chengweiou** [flutter/flutter#67871]
reproduce: (In my case)
1. 
```
setup ThemeData in MaterialApp()
  appBarTheme: AppBarTheme(
    brightness: Brightness.light,
    color: Colors.white,
    textTheme: (TextTheme(headline6: TextStyle(fontSize: 16, color: Colors.black,),)),
    iconTheme: IconThemeData(color: Colors.black),
    elevation: 1,
  ),
```
2. home page(we call it: A) no app bar
3. use onGenerateRoute to go to a another page(we call it: B)
4. in B has its appbar
```
  appBar: AppBar(
      brightness: Brightness.dark,
      iconTheme: IconThemeData(color: Colors.white,),
      backgroundColor: Colors.transparent,
      elevation: 0,
    ),
```
5. after going back. issue happens.

issue: A(actually every page)'s brightness now is Brightness.dark, theme is not working !!!
now I use B.dispose() { SystemChrome.setSystemUIOverlayStyle(SystemUiOverlayStyle.dark); } to reset A it
```
flutter doctor: 
[✓] Flutter (Channel stable, 1.22.0, on Mac OS X 10.15.7 19H2, locale en-US)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
[✓] Android Studio (version 4.0)
[!] IntelliJ IDEA Ultimate Edition (version 2020.2.1)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
[✓] Connected device (1 available)

! Doctor found issues in 1 category.
```

**Comment by TahaTesser** [comment]
Hi @chengweiou 
Does it reproduce without the App bar? See https://github.com/flutter/flutter/issues/61149
Can you please provide a complete reproducible minimal code sample
Thank you

**Comment by chengweiou** [comment]
Hi @TahaTesser

After I tested, I think maybe safearea is the problem

```
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(
        appBarTheme: AppBarTheme(
          brightness: Brightness.light,
          color: Colors.white,
          textTheme: (TextTheme(headline6: TextStyle(fontSize: 16, color: Colors.black,),)),
          iconTheme: IconThemeData(color: Colors.black),
          elevation: 1,
        ),
      ),
      onGenerateRoute: (setting) => MaterialPageRoute(
        builder: (context) {
          switch(setting.name) {
            case '/': return MyHomePage();
            case '/b': return BPage();
            default: return Text('fail');
          }
        },
      ),
    );
  }
}

class MyHomePage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      // appBar: AppBar(brightness: Brightness.light,), // add this one, it will looks fine
      body: SafeArea( // after I tested, I think this cause the issue
        child: Center(
          child: RaisedButton(
            onPressed: () => _go(context),
            child: Text('go b'),),
        ),
      ),
    );
  }
  _go(context) {
    Navigator.of(context).pushNamed('/b');
  }
}

class BPage extends StatelessWidget {
  @override build(context) => Scaffold(
    extendBody: true,
    extendBodyBehindAppBar: true,
    appBar: AppBar(
      brightness: Brightness.dark,
      iconTheme: IconThemeData(color: Colors.white,),
      backgroundColor: Colors.transparent,
      elevation: 0,
      title: Text('a'),
    ),
    body: Container(
      width: double.infinity,
      height: 300,
      // I have a image in here, so this page no safearea
      color: Colors.black,
    ),
  );
}

```

**Comment by TahaTesser** [comment]
Hi @chengweiou 
Using the code sample, I don't see any status bar resets when disposing of page B 
![ezgif com-gif-maker](https://user-images.githubusercontent.com/48603081/95962849-8a7d2180-0e24-11eb-9561-83f733f838f2.gif)

Can you please step by steps reproduce and video of the behaviour if possible 

<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel stable, 1.22.1, on Mac OS X 10.15.7 19H2, locale en-GB)
    • Flutter version 1.22.1 at /Users/tahatesser/Code/flutter_stable
    • Framework revision f30b7f4db9 (6 days ago), 2020-10-08 10:06:30 -0700
    • Engine revision 75bef9f6c8
    • Dart version 2.10.1

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.10.0.rc.1

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.50.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.15.0

[✓] Connected device (1 available)
    • RMX2001 (mobile) • EUYTFEUSQSRGDA6D • android-arm64 • Android 10 (API 29)

! Doctor found issues in 1 category.
```

</details>

**Comment by chengweiou** [comment]
@TahaTesser
...it happened....
expect: the time 1:46 should black in A, then in white in B
your video: both A,B are white....

![ezgif-3-e6c92982317c](https://user-images.githubusercontent.com/46327939/96056886-8eb53900-0e3c-11eb-961f-a8b247287599.gif)

**Comment by TahaTesser** [comment]
<details> 
<summary>complete code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      theme: ThemeData(
        appBarTheme: AppBarTheme(
          brightness: Brightness.light,
          color: Colors.white,
          textTheme: (TextTheme(
            headline6: TextStyle(
              fontSize: 16,
              color: Colors.black,
            ),
          )),
          iconTheme: IconThemeData(color: Colors.black),
          elevation: 1,
        ),
      ),
      onGenerateRoute: (setting) => MaterialPageRoute(
        builder: (context) {
          switch (setting.name) {
            case '/':
              return MyHomePage();
            case '/b':
              return BPage();
            default:
              return Text('fail');
          }
        },
      ),
    );
  }
}

class MyHomePage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      // appBar: AppBar(
      //   brightness: Brightness.light,
      // ), // add this one, it will looks fine
      body: Center(
        child: RaisedButton(
          onPressed: () => _go(context),
          child: Text('go b'),
        ),
      ),
    );
  }

  _go(context) {
    Navigator.of(context).pushNamed('/b');
  }
}

class BPage extends StatelessWidget {
  @override
  build(context) => Scaffold(
        extendBody: true,
        extendBodyBehindAppBar: true,
        appBar: AppBar(
          brightness: Brightness.dark,
          iconTheme: IconThemeData(
            color: Colors.white,
          ),
          backgroundColor: Colors.transparent,
          elevation: 0,
          title: Text('a'),
        ),
        body: Container(
          width: double.infinity,
          height: 300,
          // I have a image in here, so this page no safearea
          color: Colors.black,
        ),
      );
}

```

</details> 


<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel stable, 1.22.1, on Mac OS X 10.15.7 19H2, locale en-GB)
    • Flutter version 1.22.1 at /Users/tahatesser/Code/flutter_stable
    • Framework revision f30b7f4db9 (7 days ago), 2020-10-08 10:06:30 -0700
    • Engine revision 75bef9f6c8
    • Dart version 2.10.1

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.10.0.rc.1

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.50.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.15.0

[✓] Connected device (2 available)
    • RMX2001 (mobile)   • EUYTFEUSQSRGDA6D                     • android-arm64
      • Android 10 (API 29)
    • iPhone 11 (mobile) • 193EBBC9-CC2A-4477-B8A0-A5F12BB51400 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-0 (simulator)

! Doctor found issues in 1 category.

```

</details> 

<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel beta, 1.22.1, on Mac OS X 10.15.7 19H2, locale en-GB)
    • Flutter version 1.22.1 at /Users/tahatesser/Code/flutter_beta
    • Framework revision f30b7f4db9 (7 days ago), 2020-10-08 10:06:30 -0700
    • Engine revision 75bef9f6c8
    • Dart version 2.10.1

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.10.0.rc.1

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.50.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.15.0

 
[✓] Connected device (2 available)            
    • RMX2001 (mobile)   • EUYTFEUSQSRGDA6D                     • android-arm64
      • Android 10 (API 29)
    • iPhone 11 (mobile) • 193EBBC9-CC2A-4477-B8A0-A5F12BB51400 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-0 (simulator)

! Doctor found issues in 1 category.

```

</details> 

<details> 
<summary>flutter doctor -v</summary> 

```bash
[✓] Flutter (Channel master, 1.23.0-19.0.pre.48, on Mac OS X 10.15.7 19H2
    darwin-x64, locale en-GB)
    • Flutter version 1.23.0-19.0.pre.48 at
      /Users/tahatesser/Code/flutter_master
    • Framework revision 1133c2e85c (3 hours ago), 2020-10-15 01:57:04 -0400
    • Engine revision d2ea378be4
    • Dart version 2.11.0 (build 2.11.0-218.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.10.0.rc.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.50.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.15.0

[✓] Connected device (5 available)
    • RMX2001 (mobile)   • EUYTFEUSQSRGDA6D                     • android-arm64
      • Android 10 (API 29)
    • iPhone 11 (mobile) • 193EBBC9-CC2A-4477-B8A0-A5F12BB51400 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-0 (simulator)
    • macOS (desktop)    • macos                                • darwin-x64
      • Mac OS X 10.15.7 19H2 darwin-x64
    • Web Server (web)   • web-server                           • web-javascript
      • Flutter Tools
    • Chrome (web)       • chrome                               • web-javascript
      • Google Chrome 86.0.4240.80

• No issues found!

```

</details> 
Android seems to be related to https://github.com/flutter/flutter/issues/64001

**Comment by TahaTesser** [comment]
This seems to be framwork issue, complete details in https://github.com/flutter/flutter/issues/98065#issuecomment-1033604654

**Comment by uwumouse** [comment]
Hello, everyone. Any progress on resolving this issue? I still get it. 
I have this setup:
1. Main Screen with SliverAppBar. 
2. Settings Screen with default AppBar in SafeArea. 
3. Simple ChangeNotifer to track which theme to use. Here's how I manage current theme:  
![image](https://user-images.githubusercontent.com/61386270/167414781-2029e563-98aa-4342-898f-d42ac26f57ca.png)
4. In both light and dark theme I have these lines to set styles for StatusBar (for dark theme it's `SystemUiOverlayStyle.light` respectively)
![image](https://user-images.githubusercontent.com/61386270/167414995-82104dca-887f-44f2-947d-225f1b5b8431.png)
5. In settings screen I basically toggle the theme like this:
![image](https://user-images.githubusercontent.com/61386270/167415854-9958cfaf-87e0-4221-b6e1-dbba34f7dfd9.png)

What happens: overall theme in app does change, but not the Status Bar. The status bar only change when I go back to the Main Screen with the SliverAppBar. 
Also, when I chage theme from Dark to Light and go back to screen, the background color of Status Bar changes, but the foreground color (so Brightness) stays the same. So my Status bar just goes white and I can't see anything. 

Here's the code for the SliverAppBar I use in Main Screen (It's not all, but I think I've included the most important part):
![image](https://user-images.githubusercontent.com/61386270/167416651-f70c0eb3-db7b-4295-a81b-847ad24e9d4a.png)


And here's the code for AppBar used in Settings Screen: 
![image](https://user-images.githubusercontent.com/61386270/167416458-f501d972-99da-4193-beff-339f70c8eed2.png)

**Comment by uwumouse** [comment]
Here are real life example of my issue.
1. The Main Screen 
![image](https://user-images.githubusercontent.com/61386270/167418554-716806c2-3cca-4ad5-99e8-4043e04fa0d6.png)
2. Navigated to Settings screen
![image](https://user-images.githubusercontent.com/61386270/167418598-5a0b651e-3e84-4337-bc10-745ccedd14fb.png)
3. Switched to Light Theme
![image](https://user-images.githubusercontent.com/61386270/167418636-46ec49aa-e59f-4d52-9a30-db3cd365852a.png)
4. Navigated back to the Main Screen, Status bar changed the color, but not the icons brightness.
![image](https://user-images.githubusercontent.com/61386270/167418653-c3acbd11-3048-44a9-9ffe-b880574b664f.png)

**Comment by ImanGhasemiArani** [comment]
hi,
I have same issue about that. but i solve it.
add code below before `runApp()`.
```
SystemChrome.setSystemUIOverlayStyle(const SystemUiOverlayStyle(
    statusBarColor: Colors.transparent,
    statusBarIconBrightness: Brightness.dark,
  ));
```

then set `SystemUiOverlayStyle.dark` and `SystemUiOverlayStyle.light` for `AppBarTheme `in dark and light mode.

finally when theme changed or navigate to new page with different brightness, automatically app changed the `StatusIconBrightness`.

**Comment by exaby73** [comment]
With migrating the code to 3.7, I can no longer reproduce this issue. Therefore, I will close this issue. If you can still reproduce it, please comment with steps to reproduce and a  [complete and minimal, reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) along with the output to `flutter doctor -v`.

<details><summary><code>flutter doctor -v</code> (Master)</summary>

```
[!] Flutter (Channel master, 3.10.0-10.0.pre.39, on macOS 13.3.1 22E261 darwin-arm64, locale en-US)
    • Flutter version 3.10.0-10.0.pre.39 on channel master at /Users/nabeelparkar/fvm/versions/master
    ! Warning: `dart` on your path resolves to /opt/homebrew/Cellar/dart/2.19.5/libexec/bin/dart, which is not inside your current Flutter SDK checkout at /Users/nabeelparkar/fvm/versions/master. Consider adding /Users/nabeelparkar/fvm/versions/master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision d186792c00 (3 hours ago), 2023-04-20 20:50:59 -0700
    • Engine revision 122c3b3820
    • Dart version 3.1.0 (build 3.1.0-26.0.dev)
    • DevTools version 2.23.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at /Users/nabeelparkar/Library/Android/sdk/
    • Platform android-33, build-tools 33.0.2
    • ANDROID_SDK_ROOT = /Users/nabeelparkar/Library/Android/sdk/
    • Java binary at: /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/222.4459.24.2221.9862592/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E222b
    • CocoaPods version 1.12.0

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /Applications/Brave Browser.app/Contents/MacOS/Brave Browser

[✓] Android Studio (version 2022.1)
    • Android Studio at /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/221.6008.13.2211.9619390/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.15+0-b2043.56-8887301)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Users/nabeelparkar/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/222.4459.24.2221.9862592/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.77.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.62.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 13.3.1 22E261 darwin-arm64
    • Chrome (web)    • chrome • web-javascript • Brave Browser 112.1.50.119

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```

</details>
